src/main/java/com/tde/mescloud/service/ProductionOrderServiceImpl.java

```
@@ -177,6 +177,16 @@ public Optional<ProductionOrderEntity> findById(Long id) {
         return repository.findById(id);
     }
 
+    @Override
+    public Optional<ProductionOrderDto> findDtoById(Long id) {
+        Optional<ProductionOrderEntity> entity = repository.findById(id);
+        if(entity.isPresent()) {
```

Eu fazia if (entity.isEmpty()) e retornava o Optional.empty aqui. Para seguir aquela lógica do fail fast, tratar excepções à cabeça. Depois é que ficaria o funcionamento normal do método.